src/stages/stage5/timeline.c

```diff
@@ -400,17 +366,114 @@ static int stage5_superbullet(Enemy *e, int t) {
 	return 1;
 }
 
-void stage5_events(void) {
-	TIMER(&global.timer);
+TASK(greeter_fairy_attack, {
+		BoxedEnemy e;
+		int count;
+}) {
+	Enemy *e = TASK_BIND(ARGS.e);
+	cmplx dir = cnormalize(global.plr.pos - e->pos);
+
+	for(int i = -ARGS.count; i <= ARGS.count; i++) {
+		PROJECTILE(
+			.proto = pp_bullet,
+			.pos = e->pos,
+			.color = RGB(0.0, 0.0, 1.0),
+			.move = move_asymptotic_simple(3.5 * cdir(carg(dir) + 0.06 * i), 5),
+		);
+		play_sfx("shot1");
+	}
+}
+
+TASK(greeter_fairy, {
+		cmplx pos;
+		MoveParams move_enter;
+		MoveParams move_exit;
+}) {
+	Enemy *e = TASK_BIND(espawn_fairy_blue_box(ARGS.pos, ITEMS(.points = 1)));
+	e->move = ARGS.move_enter;
+
+	for (int i = 0; i < 5; i++) {
+		INVOKE_SUBTASK_DELAYED(80, greeter_fairy_attack, {
```

Comment:
I think you could rewrite this into a normal function instead of a task. It doesn't have any async control flow, and the delayed invocation can be replaced with just a `WAIT(80)` before the loop. The `WAIT(200)` in the end would have to be shortened to a `WAIT(20)` to match the original behavior.

With delayed invocations it doesn't match the original either by the way, because the loop already waits for `20 * 5 = 100` frames.

Either of those should match:

```c
WAIT(80);
for(int i = 0; i < 5; i++) {
	greeter_fairy_attack(...);
	WAIT(20);
}
WAIT(20);
e->move = ARGS.move_exit;
```

```c
for(int i = 0; i < 5; i++) {
	INVOKE_SUBTASK_DELAYED(80, greeter_fairy_attack, ...);
	WAIT(20);
}
WAIT(100);
e->move = ARGS.move_exit;
```

```c
for(int i = 0; i < 5; i++) {
	INVOKE_SUBTASK_DELAYED(80 + i * 5, greeter_fairy_attack, ...);
}
WAIT(200);
e->move = ARGS.move_exit;
```

I would personally choose number 1 because, as I said, the subtask is not really necessary here. You could also just inline the "attack" function.